### PR TITLE
fontconfig, bump, fix reference to libpng16/libSDL_*_1.2

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

**Comment by Begasus** [comment]
This is used by several packages so it should be looked into before merging, builds (and tests) ok on x86_gcc2 and _x86

**Comment by Begasus** [comment]
Not there yet :)

**Comment by Begasus** [comment]
This should be ok, testers welcome :)

File: media-libs/fontconfig/patches/fontconfig-2.13.1.patchset [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
++    FcChar8 *path = FcStrBuildFilename;
+ 
+     if (n)
+     {
+@@ -2251,7 +2252,7 @@ FcConfigRealFilename (FcConfig		*config,
+ 		if (!dirname)
+ 		    return NULL;
+ 
+-		FcChar8 *path = FcStrBuildFilename (dirname, buf, NULL);
++		*path = FcStrBuildFilename (dirname, buf, NULL);
```
Comment (korli) [inline media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]: This doesn't look correct. it should be:
```
path = FcStrBuildFilename (dirname, buf, NULL);

File: media-libs/fontconfig/patches/fontconfig-2.13.1.patchset [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+diff --git a/src/fccfg.c b/src/fccfg.c
+index d7c48e8..e8bbd50 100644
+--- a/src/fccfg.c
++++ b/src/fccfg.c
+@@ -2228,6 +2228,7 @@ FcConfigRealFilename (FcConfig		*config,
+     const FcChar8 *sysroot = FcConfigGetSysRoot (config);
+     FcChar8 *n = FcConfigFilename (url);
+     FcChar8 *nn = NULL;
++    FcChar8 *path = FcStrBuildFilename;
```
Comment (korli) [inline media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]: This should be moved down like this:
```
FcChar8 *dirname = FcStrDirname (nn);
FcChar8 *path;

File: media-libs/fontconfig/patches/fontconfig-2.13.1.patchset [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+diff --git a/src/fccfg.c b/src/fccfg.c
+index d7c48e8..8321d6e 100644
+--- a/src/fccfg.c
++++ b/src/fccfg.c
+@@ -2228,6 +2228,8 @@ FcConfigRealFilename (FcConfig		*config,
+     const FcChar8 *sysroot = FcConfigGetSysRoot (config);
+     FcChar8 *n = FcConfigFilename (url);
+     FcChar8 *nn = NULL;
++    FcChar8 *dirname = FcStrDirname (nn);
++    FcChar8 *path;
```
Comment (korli) [inline media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]: this should be moved in next-next block. dirname is already defined somewhere?

Comment (Begasus) [reply media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]: Moved it to the next block, now the test for 106632 also works fine, so the patch for Haiku isn't needed anymore

File: media-libs/fontconfig/patches/fontconfig-2.13.1.patchset [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+index d7c48e8..dc59eb4 100644
+--- a/src/fccfg.c
++++ b/src/fccfg.c
+@@ -2246,12 +2246,15 @@ FcConfigRealFilename (FcConfig		*config,
+ 
+ 	    if (!FcStrIsAbsoluteFilename (buf))
+ 	    {
+-		FcChar8 *dirname = FcStrDirname (nn);
++	    FcChar8 *dirname;
++        FcChar8 *path;
```
Comment (alaviss) [inline media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]:
```suggestion
+		FcChar8 *path;
```

File: media-libs/fontconfig/patches/fontconfig-2.13.1.patchset [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+diff --git a/src/fccfg.c b/src/fccfg.c
+index d7c48e8..dc59eb4 100644
+--- a/src/fccfg.c
++++ b/src/fccfg.c
+@@ -2246,12 +2246,15 @@ FcConfigRealFilename (FcConfig		*config,
+ 
+ 	    if (!FcStrIsAbsoluteFilename (buf))
+ 	    {
+-		FcChar8 *dirname = FcStrDirname (nn);
++	    FcChar8 *dirname;
```
Comment (alaviss) [inline media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]:
```suggestion
+		FcChar8 *dirname = FcStrDirname (nn);
```

File: media-libs/fontconfig/patches/fontconfig-2.13.1.patchset [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
++        FcChar8 *path;
++
++		dirname = FcStrDirname (nn);
```
Comment (alaviss) [inline media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]:
```suggestion
```

Comment (Begasus) [reply media-libs/fontconfig/patches/fontconfig-2.13.1.patchset]: Hope this one is ok :)

**Comment by korli** [comment]
Looks good to me. Why WIP?

**Comment by Begasus** [comment]
@korli this is used by many recipes and with the soname change it should be checked out with others that rely on it, so far I haven't run into any problems with it on my gcc2h install

**Comment by Begasus** [comment]
Should probably need a mass revbump then

**Comment by korli** [comment]
there is no soname change AFAICT so it should work as is.

**Comment by korli** [comment]
see https://abi-laboratory.pro/?view=timeline&l=fontconfig

**Comment by Begasus** [comment]
Than it should be ok to merge :)

**Pull request merged**